src/iiif/object_store.py

```diff
@@ -16,31 +17,68 @@ def get_object_store_connection():
     return Connection(**settings.OBJECT_STORE)
 
 
-def store_object_on_object_store(connection, local_zip_file_path, filename):
+def store_small_object_on_object_store(connection, local_zip_file_path, filename):
     with open(local_zip_file_path, 'rb') as local:
         connection.put_object(
-            settings.OBJECT_STORE_CONTAINER_NAME,
+            settings.OS_CONTAINER_NAME,
             filename,
             contents=local,
             content_type='application/zip'
         )
 
 
+def store_large_object_on_object_store(local_zip_file_path, filename):
+    with SwiftService(options=settings.OS_LARGE_FILE_OPTIONS) as swift:
+        try:
+            container = settings.OS_CONTAINER_NAME
+            objects = [SwiftUploadObject(local_zip_file_path, object_name=filename)]
+            for r in swift.upload(container=container, objects=objects):
+                if r['success']:
+                    if 'for_object' in r:
+                        # A part of the object was uploaded, so we continue to the next part
+                        continue
+                    elif 'object' in r:
+                        # The final part of the object was uploaded to the Object Store,
+                        # so here we can stop the process
+                        return
+                else:
+                    if r['action'] == "create_container":
+                        log.warning(
+                            f"Warning: failed to create container {container} {r['error']}"
+                        )
+                    elif r['action'] == "upload_object":
+                        log.error(
+                            f"Failed to upload object %s to container {container}: {r['object']}, {r['error']}"
+                        )
+                    else:
+                        log.error(r['error'])
+        except SwiftError as e:
+            log.error(e.value)
+
+
+def store_object_on_object_store(connection, local_zip_file_path, filename):
+    file_size = os.path.getsize(local_zip_file_path)
+    if file_size < settings.OS_LARGE_FILE_SIZE:
+        store_small_object_on_object_store(connection, local_zip_file_path, filename)
+    else:
+        store_large_object_on_object_store(local_zip_file_path, filename)
+
+
 def create_object_store_temp_url(connection, file_name, expiry_minutes=0, expiry_hours=0, expiry_days=0):
     # Create signature body
     method = 'GET'
     duration_in_seconds = ((((expiry_days * 24) + expiry_hours) * 60) + expiry_minutes) * 60
     expires = int(time() + duration_in_seconds)
-    path = os.path.join(f'/{settings.OBJECT_STORE_CONTAINER_NAME}', file_name)
+    path = os.path.join(f'/{settings.OS_CONTAINER_NAME}', file_name)
```

Comment:
`os.path.join('/', settings.OS_CONTAINER_NAME, file_name)`

of, nog mooier (maar werkt beter wanneer hele project pathlib.Path gebruikt, al vanuit de settings):

```
from pathlib import Path
path = Path('/') + settings.OS_CONTAINER_NAME + file_name
```